pkg/plugins/policies/meshaccesslog/api/v1alpha1/validator.go

```diff
@@ -0,0 +1,150 @@
+package v1alpha1
+
+import (
+	common_proto "github.com/kumahq/kuma/api/common/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/validators"
+	matcher_validators "github.com/kumahq/kuma/pkg/plugins/policies/matchers/validators"
+)
+
+func (r *MeshAccessLogResource) validate() error {
+	var verr validators.ValidationError
+	spec := validators.RootedAt("spec")
+
+	r.validateTop(spec, &verr)
+	r.validateFrom(spec, &verr)
+	r.validateTo(spec, &verr)
+	r.validateIncompatibleCombinations(spec, &verr)
+	r.validateToOrFromDefined(spec, &verr)
+
+	return verr.OrNil()
+}
+
+func (r *MeshAccessLogResource) validateBackend(backend *MeshAccessLog_Backend, verr *validators.ValidationError, backendIndexed validators.PathBuilder) {
+	reference := bool2int(backend.GetReference() != nil)
+	file := bool2int(backend.GetFile() != nil)
+	tcp := bool2int(backend.GetTcp() != nil)
+
+	if reference+file+tcp != 1 {
```

Comment:
Are you looking for a xor?

Reply:
XOR with 3 inputs will allow all of them to be enabled when only one of them should be allowed to be enabled, see:

![image](https://user-images.githubusercontent.com/2753650/189903488-35e61518-3618-41c8-8b07-ce9e04f41d37.png)

So I don't think we could use xor here.

Reply:
So can I leave it as it is?

Reply:
Yeah I can't find another way :(